### PR TITLE
Feat/post - AddNewPost & FetchPost

### PR DESCRIPTION
**Pull request by abhij1607**
This PR consist of

- firebase calls to add new post & fetch all post
- update in userSlice to add new post
- added postSlice to update the fetched post in store
- AddPost component
- Post component
- home page displaying AddPost and Post by peoples/users followed by user

Note: 
- there's one extra empty post, at the end, will fix that in the upcoming PR

### CODE REVIEW

**Review by FarhanMobashir — APPROVED** [review]
Clean and modular code. Consistent naming. Can be merged

File: src/components/AddPost/AddPost.jsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  Stack,
+  Button,
+  Box,
+  Avatar,
+  TextField,
+  IconButton,
+} from "@mui/material";
+
+import PhotoCameraBackIcon from "@mui/icons-material/PhotoCameraBack";
+import EmojiEmotionsIcon from "@mui/icons-material/EmojiEmotions";
```
Comment (prachi-sahani) [inline src/components/AddPost/AddPost.jsx:18]: suggestion - material UI imports can be moved to a common utilities folder

**Review by prachi-sahani — COMMENTED** [review]
Overall the code is clean and well composed
Good to merge

File: src/components/AddPost/AddPost.jsx (hunk 1 above)
Comment (abhij1607) [reply src/components/AddPost/AddPost.jsx:18]: thanks for the suggestion, I'll sure abstract them later

**Comment by abhij1607** [comment]
got 2 reviews, merging now

**Pull request merged**